docs/DesignDocs/Enabledebugmalloc.md

```diff
@@ -0,0 +1,154 @@
+
+  ```
+
+## Fine-grained runtime control
+Currently, debug malloc prints leaks only during enclave termination.
+However, users would like to isolate and track specific memory allocations.
+To achieve this, the following APIs are provided in a new published header `debugmalloc.h`.
+
+- `void oe_debug_malloc_start_tracking(void)`
+  Start tracking memory allocations.
+
+- `void oe_debug_malloc_stop_tracking(void)`
+  Stop tracking memory allocations.
+
+- `void oe_debug_malloc_print_objects(void)`
+  Print tracked objects that alive, ie not yet freed.
+
+- `void oe_debug_malloc_init_callback(void)`
+  This function will be called by debug malloc during startup.
+  Users can implement this function to start a tracking session at the earliest possible
+  time.
+
+Please note that the timing of calling of these functions is important. For
+example, the starting and stoping tracking must cover the coupled malloc/free
+operations, which means start tracking before malloc and also stop tracking
+after free, or debug malloc will definitely trigger false alarms. And by the same
+logic, oe_debug_malloc_print_objects() can only be called after tracking is
+stopped.
+
+# Implementation
+
+## Building OE SDK from source retains current behavior.
+
+OE SDK build system will retain the USE_DEBUG_MALLOC cmake parameter.
+By default, in debug build, it is set to `on` and enclaves will link against
+oe_debug_malloc and enable leak detection.
+
+The default implementation of oe_debug_malloc_init_callback will be thus:
+```c
+void oe_debug_malloc_init_callback(void)
+{
+#ifdef OE_USE_DEBUG_MALLOC // defined when USE_DEBUG_MALOC=on
+   oe_debug_malloc_start_tracking();
+#endif
+}
+```
+Of course, this default setting can be turned off by changing USE_DEBUG_MALLOC
+to `off`.
+
+## User-defined callback function to subsitute the predefined function
+While user defines the callback function following the definition and matches
+the name exactly, the predefined function will be substituted by user's, thus
+user can decide the point to start tracking by calling the callback function.
+
+The expected implementation of the callback function is as following:
+  ```c
+  void oe_debug_malloc_init_callback(void)
+  {
+      oe_debug_malloc_start_tracking();
+  }
+  ```
+
+# Sample
+
+```c
+#include <openenclave/debugmalloc.h>
+#include <openenclave/corelibc/stdlib.h>
+#include <stdio.h>
+
+// Include the trusted debugmalloc header that is generated
+// during the build. This file is generated by calling the
+// sdk tool oeedger8r against the debugmalloc.edl file.
+#include "debugmalloc_t.h"
+
+// Callback function to initiate debugmalloc
+void oe_debug_malloc_init_callback(void)
+{
+    oe_debug_malloc_start_tracking();
+}
+
+// This is the function that the host calls. It prints
+// a message in the enclave before calling back out to
+// the host to print a message from there too.
+void enclave_hello()
+{
+    oe_debug_malloc_init_callback();
```

Comment:
Delete this line since line 57 explains the callback is called automatically by the library so the app developer doesn't need to call it

Reply:
Yes now this public function is removed.